src/work/PokerHandManagement.sol

```diff
@@ -0,0 +1,123 @@
+
+    // Event to log when a hand is dealt
+    event HandDealt(uint256 indexed gameId, address player);
+
+    // Function to initialize a new deck for a game
+    function initializeDeck(uint256 gameId) internal {
+        Deck storage deck = decks[gameId]; // Get the deck for the given game ID
+        uint8 index = 0; // Initialize card index
+        for (uint8 suit = 0; suit < 4; suit++) { // Loop through all suits
+            for (uint8 rank = 2; rank <= 14; rank++) { // Loop through all ranks
+                deck.cards[index] = Card(rank, suit); // Assign rank and suit to the card
+                index++; // Increment card index
+            }
+        }
+        deck.topCard = 0; // Reset the top card index
+    }
+
+    // Function to shuffle the deck for a game
+    function shuffleDeck(uint256 gameId) internal {
+        Deck storage deck = decks[gameId]; // Get the deck for the given game ID
+        for (uint8 i = 51; i > 0; i--) { // Loop through the deck in reverse order
+            uint8 j = uint8(uint256(keccak256(abi.encodePacked(block.timestamp, block.difficulty, i))) % (i + 1)); // Generate a random index
+            Card memory temp = deck.cards[i]; // Swap the current card with the random card
+            deck.cards[i] = deck.cards[j];
+            deck.cards[j] = temp;
+        }
+    }
+
+    // Function to deal a hand to a player
+    function dealHand(uint256 gameId, address player) internal {
+        Deck storage deck = decks[gameId]; // Get the deck for the given game ID
+        require(deck.topCard <= 50, "Not enough cards in deck"); // Ensure there are enough cards left in the deck
+
+        Hand memory hand; // Create a new hand
+        hand.cards[0] = deck.cards[deck.topCard]; // Deal the first card
+        deck.topCard++; // Move to the next card
+        hand.cards[1] = deck.cards[deck.topCard]; // Deal the second card
+        deck.topCard++; // Move to the next card
+
+        bytes memory encryptedHand = encryptHand(gameId, player, hand); // Encrypt the dealt hand
+        encryptedHands[gameId][player] = encryptedHand; // Store the encrypted hand
+
+        emit HandDealt(gameId, player); // Emit the HandDealt event
+    }
+
+    // Function to encrypt a hand
+    function encryptHand(uint256 gameId, address player, Hand memory hand) internal returns (bytes memory) {
+        string memory handString = string(abi.encodePacked(
+            cardToString(hand.cards[0]), ",",
+            cardToString(hand.cards[1])
+        )); // Convert the hand to a string
+
+        bytes memory encryptedHand = LitAccessControl.encrypt(
+            bytes(handString),
+            address(this),
+            gameId,
+            player
+        ); // Encrypt the hand string
+
+        return encryptedHand; // Return the encrypted hand
+    }
+
+    // Function to get the hand for the caller
+    function getHand(uint256 gameId) external view returns (string memory) {
```

Comment:
we can create a role that gets granted to the address when its their turn and so that way we dont have to do the msg.sender check